### ISSUE TITLE
Weird Album Sorting on Artist Page

### ISSUE THREAD

**Issue by TinDefacto** [jmshrv/finamp#536]
Hi there!

Some recent update -- I'm not sure which one -- sent the sorting of albums on an artist's page all out of wack. Now the order of an artist's releases may be (as in one case for me) 2021, 2016, 2019, 2013, 2020, 2013, 2011, 2014, 2019, 2016, 2022. It's not even alphabetical, so I'm really confused! I really appreciated the previous chronological order, and I don't see this issue being raised elsewhere, so I figured I'd point it out. 

Thanks again for all the work y'all do. :)

**Comment by Chaphasilor** [comment]
That might've been https://github.com/jmshrv/finamp/pull/448 or #493

Could you check the full metadata in Jellyfin to see what the PremiereDate of these albums is, compared to the production year?

**Comment by TinDefacto** [comment]
Oh, interesting -- thanks for the response! I hadn't seen #448, and that
looks most likely to be the culprit, but when I look at the metadata on
Jellyfin, literally only one of the albums has a release date, and that's
the 2022 one (which appears last). So I'm not sure what's going on?

**Comment by TinDefacto** [comment]
And doing a quick survey of other artists, it seems like it's correctly
sorting albums that have release dates filled in -- but for those that only
have a year, I can't find rhyme or reason to the order it decides on.

**Comment by Chaphasilor** [comment]
If there is no release date provided, the order is most likely just random, as you are observing. We should fall back to sorting by production year and then name I think

**Comment by TinDefacto** [comment]
Yes, I think that would make the most sense! Preferably by taking any blank
release dates to be YEAR-01-01, so it gets sorted in with the dates ones,
if that makes sense.

**Comment by Maxr1998** [comment]
I remember doing two PRs for this issue, since some changes were undone in between. Maybe that's just an artifact of that, and not actually an issue? @TinDefacto could you verify whether it's still an issue on the latest version?

**Comment by TinDefacto** [comment]
Hi! Okay, thanks so much for your work on this -- though I think it still
needs a bit more work!

Some artists on my server (like those that have multiple releases per year)
have full release dates for each album -- these are now displayed in
perfect chronological order, which is immaculate.

However! For other artists I either have a mix of exact release dates or
blank ones (i.e. just the release year) or only years for each release, no
exact release date. These it seems to just be putting in alphabetical order
by album name, which is very much not what I want.

What I feel might be a good solution is to, for albums with only years and
no release date, sort it by release date by filling in the missing ones
with YYYY-01-01 (or something similar). Does that seem reasonable/doable?

Thank you again!!

**Comment by Maxr1998** [comment]
As the sorting is done on the server side (except when offline), I suppose it should be fixed there instead. But thanks for reporting it, I'll give it a look when I find some time.
